https.go

```diff
@@ -334,6 +346,22 @@ func (proxy *ProxyHttpServer) handleHttps(w http.ResponseWriter, r *http.Request
 	}
 }
 
+func createCustomConnectResponse(ctx *ProxyCtx) ([]byte, error) {
+	if ctx.proxy.ConnectRespHandler == nil {
+		return nil, nil
+	}
+	resp := &http.Response{Status: "200 OK", StatusCode: 200, Proto: "HTTP/1.0", Header: http.Header{}}
+	err := ctx.proxy.ConnectRespHandler(ctx, resp)
```

Comment:
if we want the response to contain any info from the connection to the backend server, do we also need to pass it the `net.Conn` (`targetSiteCon`)?

Reply:
Should be available on the context! e.g. to get destination IP it should be possible to get the smokescreen context from the goproxy context UserData like [this](https://github.com/stripe/smokescreen/blob/14a7af2d8f79b9a8399e81a571a8e70d3ca2d95e/pkg/smokescreen/smokescreen.go#L249), and then get decision.resolvedAddr from that.

Reply:
oooooh nice!